src/circuits/babyjub.rs

```diff
@@ -197,22 +197,22 @@ impl<F: FieldExt> AltJubChip<F> {
             ],
             [None, None, None, None, Some(-F::one()), None, Some(F::one()), None, None],
             0
-        )?[4].clone();
-        let lambda = self.config.assign_line(region, &mut (), offset,
+        )?[2].clone();  // verify y1y2 is correct
+        let d_lambda = self.config.assign_line(region, &mut (), offset,
             [
                 Some(lambda1),
                 None,
                 None,
                 Some(lambda2),
-                Some(Limb::new(None, y1y2 * x1x2)),
+                Some(Limb::new(None, d*y1y2 * x1x2)),
```

Comment:
need to calculate d*y1y2*x1x2 in circuits, not in witness.

Reply:
It's correct, but I hope this project can be format.